contracts/dao/voting/lockdrop-vault/src/contract.rs

```diff
@@ -252,6 +312,91 @@ pub fn query_bonding_status(
     })
 }
 
+fn get_voting_power_for_address(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    address: String,
+    height: u64,
+) -> StdResult<Decimal256> {
+    get_voting_power(
+        deps,
+        config,
+        pool_type,
+        &LockdropQueryMsg::QueryUserLockupTotalAtHeight {
+            pool_type,
+            user_address: address,
+            height,
+        },
+        height,
+    )
+}
+
+fn get_voting_power_total(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    height: u64,
+) -> StdResult<Decimal256> {
+    get_voting_power(
+        deps,
+        config,
+        pool_type,
+        &LockdropQueryMsg::QueryLockupTotalAtHeight { pool_type, height },
+        height,
+    )
+}
+
+fn get_voting_power(
+    deps: Deps,
+    config: Config,
+    pool_type: PoolType,
+    msg: &LockdropQueryMsg,
+    height: u64,
+) -> StdResult<Decimal256> {
+    let lp_tokens: Option<Uint128> = deps
+        .querier
+        .query_wasm_smart(config.lockdrop_contract, msg)?;
+
+    let lp_tokens = Decimal256::new(Uint256::from(lp_tokens.unwrap_or_default()));
+
+    if lp_tokens.is_zero() {
+        return Ok(Decimal256::zero());
+    }
+
+    let asset_denom = match pool_type {
+        PoolType::ATOM => "uatom",
+        PoolType::USDC => "usdc",
+    };
+
+    let twap: Vec<(AssetInfo, Decimal256)> = deps.querier.query_wasm_smart(
+        config.oracle_contract,
+        &OracleQueryMsg::TWAPAtHeight {
+            token: AssetInfo::NativeToken {
+                denom: asset_denom.to_string(),
+            },
+            height: Uint64::new(height),
+        },
+    )?;
+
+    let untrn_twap_assets = twap
+        .iter()
+        .filter(|x| {
+            let (asset, _amount) = x;
+
+            match asset {
+                AssetInfo::NativeToken { denom } => denom == asset_denom,
+            }
+        })
+        .map(|x| x.1);
+
+    let total_twap_untrn = untrn_twap_assets
+        .fold(Decimal256::zero(), |acc, amount| acc + amount)
+        .sqrt();
```

Comment:
could you please clarify why `sqrt(sum(asset_amounts...))` here?

Reply:
Because at the moment I do not have clear vision what kind of result will be returned from oracle contract. But anyway it will return array as the result as far as I understand the code. But this part will be improved after integrational tests will be completed.